### DISCUSSION TITLE
value caching on qrcode

### DISCUSSION THREAD

**Discussion by bkbartk** [Blackymas/NSPanel_HA_Blueprint#2189] [Q&A] [answered]
I'm trying to create a wifi network with a daily changing password.
So I created an input text which is updated by an automation and use this one in the blueprint
```
      qrcode_value: WIFI:T:WPA2;S:wifi-guest;P:{{ states('input_text.guestwifi') }};;
      qrcode_label: 'WIFI: wifif-guest {{ states(''input_text.guestwifi'') }}'
```
the issue here is that the value on the nspanel only updates if I restart all automations.
Is there a way to refresh those texts without having to restart the autmation?

**Comment by edwardtfn** [comment, marked as answer]
I’ve never tried this, but you can try using the [QR code service](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/api.md#qr-code-service-qrcode).
This service is called every time the panel restarts or when Home Assistant restarts. You can leave que QR code field empty in the blueprint automation related to that panel and create another automation triggering a few seconds after those two things, calling this service with the content of your input text entity…
The QR code is currently stored in Nextion, so it must be sent with every boot.

I have a very old PR to support dynamic QR code, but that was rejected (😂). It would be easy to update that to latest code: https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/504

**Reply by edwardtfn** [reply]
Related:
- #452
- #504

**Comment by bkbartk** [comment]
thnxs that's brilliant,
and don't need it to be dynamic as long as I can update it using an automation.
Keeping the settings also in the blueprint makes sure the qr still works if something gets restarted.
